### PR TITLE
Init flow frameworks cli.

### PR DESCRIPTION
**Pull request by svnsairam**
### Description

CLI for framework stack.

Reference PR: https://github.com/firebase/firebase-tools/pull/5846

### CODE REVIEW

File: src/commands/init.ts [hunk 1]
```diff
@@ -71,6 +72,15 @@ const choices = [
     checked: false,
   },
 ];
+
+if (isEnabled("internaltesting")) {
```
Comment (taeold) [inline src/commands/init.ts:76]: please create a new experiment (e.g. `frameworks`). `internaltesting` is for another experiment.

File: src/commands/init.ts [hunk 2]
```diff
@@ -71,6 +72,15 @@ const choices = [
...
+  choices.push({
+    value: "frameworkstack",
```
Comment (taeold) [inline src/commands/init.ts:78]: let's use the name `frameworks` more consistently. I seem `frameworkstack` and `frameworkstackapi` and would prefer seeing `frameworks`

File: src/init/features/frameworkstack/constants.ts [hunk 3]
```diff
@@ -0,0 +1,15 @@
+export const DEFAULT_REGION = "us-central1";
+export const ALLOWED_REGIONS = [
+  { name: "us-east1 (South California)", value: "us-east1" },
+  { name: "us-east4 (Northern Virginia)", value: "us-east4" },
+  { name: "us-east5 (Columbus)", value: "us-east5" },
+  { name: "us-central1 (Iowa)", value: "us-central1" },
+  { name: "us-central2 (Chicago, IL)", value: "us-central2" },
+  { name: "us-south1 (Dallas)", value: "us-south1" },
+  { name: "us-west1 (Oregon)", value: "us-west1" },
```
Comment (taeold) [inline src/init/features/frameworkstack/constants.ts:9]: Where did we get this list?

If we are making this up, then I'd just pick `us-central1` only

File: src/init/features/frameworkstack/index.ts [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+  setup.frameworkstack = {};
+
+  utils.logBullet("First we need a few details to create your service.");
+
+  await promptOnce(
+    {
+      name: "serviceName",
+      type: "input",
+      default: "acme-inc-web",
+      message: "Create a name for your service [1-64 characters]",
```
Comment (taeold) [inline src/init/features/frameworkstack/index.ts:37]: It looks like we aren't validating this requirement at all. Can we validate it and reprompt if it doesn't meet our requirement?

File: src/init/index.ts [hunk 5]
```diff
@@ -29,6 +29,7 @@ const featureFns = new Map<string, (setup: any, config: any, options?: any) => P
   ["project", features.project], // always runs, sets up .firebaserc
   ["remoteconfig", features.remoteconfig],
   ["hosting:github", features.hostingGithub],
+  ["frameworkstack", features.frameworkstack],
```
Comment (taeold) [inline src/init/index.ts:32]: This chioce has to be inside an expeirment. We don't want this choice to show up for most users.

File: src/init/features/frameworkstack/constants.ts (hunk 3 above)
Comment (svnsairam) [reply src/init/features/frameworkstack/constants.ts:9]: This was according to doc shared here: https://www.figma.com/file/soCklCIUh3XVQQZPQm9Olx/Firestack?type=design&node-id=1004-6227&t=q4xvLZwgtAsR9hbn-0

File: src/init/features/frameworkstack/index.ts (hunk 4 above)
Comment (svnsairam) [reply src/init/features/frameworkstack/index.ts:37]: The reason to leave this is, this validation should happen on backend, similar to 
the case where a new `project id` is taken as input (it should be between 6 to 32 characters) and the error is shown when the `Creating Google Cloud Platform project` step is executed. WDYT?

File: src/commands/init.ts (hunk 1 above)
Comment (svnsairam) [reply src/commands/init.ts]: done

File: src/commands/init.ts (hunk 2 above)
Comment (svnsairam) [reply src/commands/init.ts]: done

File: src/init/index.ts (hunk 5 above)
Comment (svnsairam) [reply src/init/index.ts]: done

File: src/init/features/frameworks/index.ts [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+import { isEnabled } from "../../../experiments";
+
+/**
+ * Setup new frameworks project.
+ */
+export async function doSetup(setup: any, config: Config, options: Options): Promise<void> {
+  const projectId = setup?.rcfile?.projects?.default;
+  if (projectId && !isEnabled("frameworks")) {
+    await requirePermissions({ ...options, project: projectId });
+    await Promise.all([ensure(projectId, "placeholder.googleapis.com", "unused", true)]);
```
Comment (taeold) [inline src/init/features/frameworks/index.ts:24]: nit* we might want to comment this out for now? Does this actually work?

**Review by taeold — APPROVED** [review]

File: src/init/features/frameworks/index.ts (hunk 6 above)
Comment (svnsairam) [reply src/init/features/frameworks/index.ts:24]: The line to get the `projectId` works, but the we can remove `if` condition for now. The condition was using in other places.

**Pull request merged**